flow/Arena.cpp

```diff
@@ -776,6 +776,23 @@ TEST_CASE("/flow/Arena/Size") {
 	return Void();
 }
 
+// Test that x.dependsOn(x) works, and is effectively a no-op.
+TEST_CASE("/flow/Arena/SelfRef") {
+	Arena a;
+
+	Standalone<StringRef> s = makeString(100, a);
```

Comment:
`s.arena() != a` here - makeString(int, Arena&) returns a StringRef, which gets copied into a brand new arena when converted to a Standalone<StringRef>

Reply:
Alright, I've sorted the test out and don't rely on `Standalone<StringRef>` or even `makeString`, just a plain `Arena a(4096);` which is enough to populate it.